### PR TITLE
Enhancements to Release Bundle Create Command: Optional Flags and Fallback Support

### PR DESCRIPTION
**Pull request by oshratZairi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.

---
Improvements to the release bundle create command:
- Made the `spec` flag optional, with `build-name` and `build-number` flags as replacements.
- Added support for `JFROG_CLI_BUILD_NAME` and `JFROG_CLI_BUILD_NUMBER` as fallback options when the `build-name` and `build-number` flags are not provided.

documentation - https://github.com/jfrog/documentation/pull/173/files

### CODE REVIEW

File: utils/cliutils/commandsflags.go [hunk 1]
```diff
@@ -579,6 +579,8 @@ const (
 	ReleaseBundles       = "release-bundles"
 	lcReleaseBundles     = lifecyclePrefix + ReleaseBundles
 	SigningKey           = "signing-key"
+	BuildName            = "build-name"
+	BuildNumber          = "build-number"
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:583]: No need to define the above two consts.
You should reuse the existing `buildName` and `buildNumber` consts defined above.

File: utils/cliutils/utils.go [hunk 2]
```diff
@@ -415,6 +415,10 @@ func GetSpec(c *cli.Context, isDownload, overrideFieldsIfSet bool) (specFiles *s
 	return
 }
 
+func CreateSpecFromBuildNameAndNumber(buildName string, buildNumber string) (specFiles *speccore.SpecFiles, err error) {
```
Comment (eyalbe4) [inline utils/cliutils/utils.go:418]: This wrapper function seema redundant. The `speccore.CreateSpecFromBuildNameAndNumber` can be called directly.

File: utils/tests/utils.go [hunk 3]
```diff
@@ -693,3 +693,15 @@ func createFlagSet(t *testing.T, flags []string, args []string) *flag.FlagSet {
 	assert.NoError(t, flagSet.Parse(cmdFlags))
 	return flagSet
 }
+
+func ExtractBuildInfo(build string) (string, string, error) {
```
Comment (eyalbe4) [inline utils/tests/utils.go:697]: You should not recreate this split logic.
JFrog CLI already splits the build value when it actually uses the spec file to download files. You should find that utility and reuse it. Note that code may be in jfrog-client-go or jfrog-cli-core.

File: lifecycle_test.go [hunk 4]
```diff
@@ -167,7 +169,24 @@ func TestLifecycleFullFlow(t *testing.T) {
 		// Verify the artifacts were distributed correctly by the provided path mappings.
 		assertExpectedArtifacts(t, tests.SearchAllDevRepo, tests.GetExpectedLifecycleDistributedArtifacts())
 	*/
+}
+
+func TestCreateBundleWithoutSpec(t *testing.T) {
+	cleanCallback := initLifecycleTest(t, artifactoryLifecycleMinVersion)
+	defer cleanCallback()
+
+	lcManager := getLcServiceManager(t)
```
Comment (eyalbe4) [inline lifecycle_test.go:178]: Better to move this function call a few lines below, closer to where it is needed.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/tests/utils.go (hunk 3 above)
Comment (oshratZairi) [reply utils/tests/utils.go:697]: the logic here is just for the test - I need to add the --build-name, --build-number to the command according to the test files and then will run the cli command. The create command will run as expected with no external locig.

File: utils/cliutils/commandsflags.go [hunk 5]
```diff
@@ -164,8 +164,8 @@ const (
 	specVars = "spec-vars"
 
 	// Build info flags
-	buildName   = "build-name"
-	buildNumber = "build-number"
+	BuildName   = "build-name"
+	BuildNumber = "build-number"
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:168]: Why did we make these variables public? They are used only in this file?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: lifecycle/cli.go [hunk 6]
```diff
@@ -131,21 +134,56 @@ func validateCreateReleaseBundleContext(c *cli.Context) error {
...
+		return err
+	}
+
+	if err := validateCreationValuesPresence(c, methodCount); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateSingleCreationMethod(methodCount int) error {
```
[the remark this answers is not in the document]
Comment (oshratZairi) [reply lifecycle/cli.go:156]: this function verifies that the user didn't send  2 sources for the bundle creation for example --spec and --builds
the methodCount isn't redundant - the value for the arguments comes from the check in line 146.

[the remark this answers is not in the document]
Comment (eyalbe4) [reply lifecycle/cli.go:156]: :+1:

File: utils/cliutils/commandsflags.go (hunk 5 above)
Comment (oshratZairi) [reply utils/cliutils/commandsflags.go:168]: You have asked me to use BuildName and BuildNumber and not to add new ones to the command - so had to turn the current into public

Comment (eyalbe4) [reply utils/cliutils/commandsflags.go:168]: Got it. I didn't notice those flags are used from within a different package,

**Pull request merged**